collector/uname.go

```diff
@@ -0,0 +1,65 @@
+// Copyright 2019 The Prometheus Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// +build freebsd linux
+// +build !nouname
+
+package collector
+
+import (
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+var unameDesc = prometheus.NewDesc(
+	prometheus.BuildFQName(namespace, "uname", "info"),
+	"Labeled system information as provided by the uname system call.",
+	[]string{
+		"sysname",
+		"release",
+		"version",
+		"machine",
+		"nodename",
+		"domainname",
+	},
+	nil,
+)
+
+type unameCollector struct{}
+type unameOutput map[string]string
```

Comment:
I would prefer to make this a struct.
```
type uname struct {
  SysName string,
  Release string,
...
}

Reply:
Thanks, that makes much more sense. Not sure what I was thinking.